website/content/docs/pipelines/index.mdx

```diff
@@ -0,0 +1,438 @@
+  step "build" {
+    use "build" {}
+  }
+
+  step "my-deploy" {
+    #  "pipeline" is a special "built-in" step that references another pipeline
+    use "pipeline" {
+      project = "my-pipeline-project"
+      name    = "deploy"
+    }
+  }
+}
+
+pipeline "deploy" {
+  step "deploy" {
+    use "deploy" {}
+  }
+  step "release" {
+    use "release" {
+      prune = true
+    }
+  }
+}
+```
+
+### Pipelines Across Workspaces
+
+Pipelines can be defined and scoped to a Workspace for deploying in various environments.
+When you use the pipeline run CLI, you can run a pipeline in a different workspace
+with the `-workspace` flag.
+
+```shell
+$ workspace pipeline run builder -workspace=dev
+```
+
+#### Workspace Scoped Steps
+
+You can override the environment that a Pipeline step runs in by specifying a
+`workspace` param for any step in your Pipeline definition. For example,
+this pipelines "build-prod" step will run in the production workspace.
+
+Notice that in the app definition, it contains two build stanzas, one of which
+is _workspace scoped_. That means when the "build-prod" step runs, it will
+use that workspace scoped build plugin pack rather than the default "docker" plugin.
+
+```hcl
+// waypoint.hcl
+pipeline "builder" {
+  step "build-dev" {
+    use "build" {}
+  }
+  step "build-prod" {
+    workspace = "prod"
+    use "build" {}
+  }
+  step "deploy-dev" {
+    use "deploy" {
+      release = false
+    }
+  }
+}
+
+// waypoint.hcl
+app "tetris" {
+  build {
+    use "docker" {}
+
+    workspace "prod" {
+      use "pack" {}
+    }
+  }
+  deploy {
+    use "docker" {}
+
+    workspace "prod" {
+      use "kubernetes" {}
+    }
+  }
+}
+```
+
+You can also use this to set a workspace for an embedded pipeline:
+
+```hcl
+pipeline "builder" {
+  step "build" {
+    use "build" {}
+  }
+  step "deploy-test" {
+    workspace = "test"
+    pipeline "deploy-test-env" {
+      step "build" {
+        // test env uses docker-pull
+        use "build" {}
+      }
+      step "deploy" {
+        use "deploy" {}
+      }
+      // more steps here...
+    }
+  }
+  step "deploy-prod" {
+    workspace = "prod"
+    pipeline "deploy-prod-env" {
+      step "build" {
+        // prod env uses docker-pull
+        use "build" {}
+      }
+      step "deploy" {
+        use "deploy" {}
+      }
+      // more steps here...
+    }
+  }
+}
+```
+
+## Examples
+
+Below is an example that redefines the Up operation to be a pipeline with three
+steps for a build, deploy, and release. This example could be used as a starting
+point for more complicated pipelines.
+
+```hcl
+project = "my-project"
+
+pipeline "full-up" {
+  step "my-build" {
+    use "build" {}
+  }
+
+  step "my-deploy" {
+    use "deploy" {}
+  }
+
+  step "my-release" {
+    use "release" {}
+  }
+}
+
+app "my-app" {
+  build { ... }
+  deploy { ... }
+  release { ... }
+}
+```
+
+For the `exec` steps, it accepts a command and arg parameter for running commands.
```

Comment:
```suggestion
The `exec` step accepts "command" and "arg" parameters for executing commands.
```

Reply:
Or we can use `command` and `arg`. In fact, we should probably standardize that.